pipelines/api-pipelines.yml

```diff
@@ -20,137 +22,57 @@ pr:
-            displayName: Build
-            inputs:
-              command: build
-              projects: '**/*.csproj'
-              arguments: '--configuration Release'
-
-          - task: DotNetCoreCLI@2
-            displayName: Unit Tests
-            inputs:
-              command: test
-              projects: '**/LinkyLink.Tests/*.csproj'
-              arguments: -c $(BuildConfiguration) --collect:"XPlat Code Coverage"
-
-            # Generate the report using ReportGenerator (https://github.com/danielpalme/ReportGenerator)
-          - script: |
-              dotnet tool install -g dotnet-reportgenerator-globaltool
-              reportgenerator -reports:$(Agent.TempDirectory)/**/coverage.cobertura.xml -targetdir:$(Build.SourcesDirectory)/coverlet/reports -reporttypes:"Cobertura"
-            displayName: Create Code coverage report
-
-          - task: PublishCodeCoverageResults@1
-            displayName: 'Publish code coverage'
-            inputs:
-              codeCoverageTool: Cobertura
-              summaryFileLocation: $(Build.SourcesDirectory)/coverlet/reports/Cobertura.xml 
-              failIfCoverageEmpty: true 
+      - template: templates/api-build.yml
 
-          - task: BuildQualityChecks@6
-            displayName: 'Build quality check'
-            inputs:
-              checkCoverage: true
-              coverageFailOption: 'fixed'
-              coverageType: 'lines'
-              coverageThreshold: '25'
-
-          - task: DotNetCoreCLI@2
-            displayName: Publish packages
-            inputs:
-              command: publish
-              arguments: '--configuration $(BuildConfiguration) --output $(Build.ArtifactStagingDirectory)'
-              zipAfterPublish: True
-
-          - task: PublishBuildArtifacts@1
-            condition: and(succeeded(), ne(variables['Build.Reason'], 'PullRequest'))
-            displayName: Publish Artifacts
-            inputs:
-              pathtoPublish: '$(Build.ArtifactStagingDirectory)' 
-              artifactName: 'drop'
-              publishLocation: 'Container'
-
-  - stage: DeployStage
-    dependsOn: BuildStage
+  - stage: Development
+    displayName: Development
+    dependsOn: BuildAndValidate
     condition: and(succeeded(), ne(variables['Build.Reason'], 'PullRequest'))
-    displayName: Deploy
+    variables:
+      - group: prodoh-urlist-dev
     jobs:
-      - job: RegionalMaxtrixDeployment
-        displayName: API to Region
-        strategy:
-          matrix:
-            westus2:
-              region: westus2
-            eastus:
-              region: eastus
-            centralus:
-              region: centralus
-        steps:
-          - task: UseDotNet@2
-            displayName: Dotnet version 3.1.100
-            inputs:
-              version: '3.1.100'
+      - template: templates/api-deploy-environment.yml
+        parameters:
+          subscription: "PELITTLE TEAM - CSE DWR(d36d0808-a967-4f73-9fdc-32ea232fc81d)"
```

Comment:
I believe you had an ENV variable for subscription. (Similar comments below)

Reply:
Yes we did.  I need to review this with some of you to fix this.  There is some issue with the subscription getting evaluated when passed in through the template very strange.

Reply:
@aluong helped figure out a workaround.